### PR TITLE
Self-serve usability updates

### PR DESCRIPTION
**Pull request by djcorbett96**
Proposing the following changes to make it easier for anyone to get up and running with the consulting starter.

### CODE REVIEW

**Comment by djcorbett96** [comment]
Just pushed a new commit to delete `features.json` and add it to `.gitignore`. The dev command generates this automatically and having a features.json commited can cause deployment issues in the platform described [here](https://yext.slack.com/archives/C05T6QDDLJF/p1697572815871659).

File: README.md [hunk 1]
```diff
@@ -2,31 +2,96 @@
...
+   npm install
+   ```
+
+2. Run `yext init` and authenticate the CLI with your Yext credentials. Add the `-u sandbox` flag if you're using a sandbox Yext account.
+3. Run `yext resources apply platform-config/config` to apply the required configuration resources to your account.
+4. Run `yext resources apply platform-config/entities` to add the necessary entities to your account. This includes a test location entity which can be deleted before running the apply command if desired.
+
+   - You will be prompted to enter several API keys during this step which power the `<Reviews>`, `<Nearby>` and `<Locator>` components. If you do not plan to use these components, you can skip the API key input by pressing Enter.
+   - The Reviews API key should come from a developer app with Read access to the Management API > Reviews endpoint.
+   - For both the Nearby API key as well as the Search API key, you can use an API key from any search experience in your account or a developer app with Read access to the Content Delivery API > Entities endpoint.
```
Comment (bhainesva) [inline README.md:33]: I didn't realize you could use any key with entities access for your search api requests? That seems like something we maybe shouldn't suggest even if it technically works

File: platform-config/slug-manager/plugin/mod.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const { webhook, connector } = createSlugManager({
+  apiKey: API_KEY,
+  slugFormat:
+    "[[localeCode]]/[[address.region]]/[[address.city]]/[[address.line1]]",
+  slugFormatLocaleOverrides: {
+    en: "[[address.region]]/[[address.city]]/[[address.line1]]",
+  },
+  entityTypes: ["location"],
+  env: "sandbox",
```
[the remark this answers is not in the document]
Comment (djcorbett96) [reply platform-config/slug-manager/plugin/mod.ts:13]: Ah yes, meant to revert that

File: README.md [hunk 3]
```diff
@@ -2,31 +2,96 @@
...
+   - These keys will live on the Site entity that is created during this step, so you can always update or add the keys later once the entity has been created.
+   - For guidance on creating a developer app in the Yext plaform, see the first two steps of the Slug Manager section below.
+
+5. If you are implementing a locator search experience, run `yext resources apply platform-config/search` to apply the locator search config to your account.
+6. In the Yext platform, run the directory manager named “Directory” that was created as part of the resource apply command in Step 3. This will create the location-related entities necessary to render a directory on the frontend based on your account’s location entities.
+   1. Navigate to **Pages** > **Directory Manager**.
+   2. Click to view the directory, then click **Run**.
+7. Back in your repo, remove any fields from the location stream defined in `src/layouts/entity.tsx` that do not exist on your location entities. Failing to delete fields that don’t exist on your entities will lead to errors during development.
```
[the remark this answers is not in the document]
Comment (djcorbett96) [reply README.md:41]: The fields will be in the account, they just might not have the fields filled out on their existing location entities. I guess shouldn't cause errors though, so step 7 may be unnecessary to include now that we've commented out the product, event, etc fields by default

File: README.md (hunk 1 above)
Comment (djcorbett96) [reply README.md:33]: Yea that's fair, I can separate the two and say that the Search key should come from the search experience

**Comment by djcorbett96** [comment]
Yes I can add another subfolder of the platform-config for the "advanced options", similar to what I did for search and the base entities, that applies all of the product, professional, faq, and event entities along with the necessary config

**Comment by djcorbett96** [comment]
Added support for advanced config and addressed the other comments.

**Review by bhainesva — APPROVED** [review]

**Pull request merged**